src/Service.gs

```diff
@@ -430,14 +430,15 @@ Service_.prototype.fetchInternal_ = function(url, params, opt_token,
   }
   switch (this.paramLocation_) {
     case 'auth-header':
-      params.headers = _.extend({}, params.headers,
-          signer.toHeader(oauthParams));
+      params.headers =
+        extend_(params.headers || {}, signer.toHeader(oauthParams));
```

Comment:
Slightly unsafe change mutating params.headers directly instead of copying to a new object. 

e.g.
```
const JSON_HEADERS = {
  'content-type': 'application/json'
};

function doSomething() {
  var params = {
    header: JSON_HEADERS
  };
  var result = service.fetch('https://example.com', params);
   ...
}
```

Even though the code was already mutating params, previously that would still hve been OK as the nested objects were cloned. But new code would end up mutating the JSON_HEADERS object which could then break subsequent uses.

Ideally should treat the entire params object as immutable and make a deep copy of it so there are no side effects.

Reply:
Ya, it's something we were doing in the OAuth2 lib and I ported it over, but it is less safe. Switched to a MDN Polyfill for `Object.assign` that allows for multiple sources and using an empty target object to do a clone.